docs/content/seo.md

```diff
@@ -0,0 +1,200 @@
+
+### Generating a sitemap
+
+Google <a href="https://developers.google.com/search/docs/advanced/sitemaps/build-sitemap" target="_blank">expects the standard sitemap protocol to be implemented</a>. Thankfully, Bartholomew automatically builds a sitemap file based on the entire set of content in the CMS. The heavy lifting of the work is performed using the <a href="https://rhai.rs/" target="_blank">Rhai</a> scripting language. Here is an example of the `sitemap.rhai` file that you would store in Bartholomew's scripts directory.
+
+```
+// This function lists all of the posts, filtering a few.
+//
+// It returns an array of objects of the form:
+//  [
+//    #{ uri: "path/to/page", page: PageObject }
+// ]
+
+// These should be skipped.
+let disallow = [
+    "/sitemap", // Don't list self.
+    // "/tag", // tag will list all of the tags on a site. If you prefer this not be indexed, uncomment this line.
+    "/index", // This is a duplicate of /
+    "/atom",
+    "/robots",
+];
+
+// Param 1 should be `site.pages`
+let pages = params[0];
+
+let site_pages = [];
+let keys = pages.keys();
+for item in keys {
+    let path = item.sub_string(8);
+    let page = pages[item];
+
+    path = path.sub_string(0, path.index_of(".md"));
+    if !disallow.contains(path) {
+        site_pages.push(#{
+            uri: path,
+            page: page,
+            priority: prioritize(path),
+            frequency: "weekly",
+        });
+    }
+   
+}
+
+// This is an example of how we could prioritize based on information about the page.
+//
+// Specifically, here we use path to boost docs and blogs while reducing the priority
+// of author pages and features.
+fn prioritize(path) {
+    let boost = ["/blog/", "/docs/"];
+    for sub in boost {
+        if path.contains(sub) {
+            return 0.8
+        }
+    }
+    let nerf = ["/author/", "/features/"];
+    for sub in nerf {
+        if path.contains(sub) {
+            return 0.3
+        }
+    }
+    0.5
+}
+
+// Return the blogs sorted newest to oldest
+fn sort_by_date(a, b) {
+    if a.page.head.date < b.page.head.date {
+        1
+    } else {
+        -1
+    }
+}
+
+// Sort by the value of the page date.
+site_pages.sort(Fn("sort_by_date"));
+
+site_pages
+```
+
+In conjunction to the above scripting, the aforementioned <a href="https://handlebarsjs.com/" target="_blank">Handlebars</a> templating assists in this work being performed dynamically (using variables common between the script and the template); as shown in the `sitemap.hbs` file's contents below.
+
+```
+<?xml version="1.0" encoding="UTF-8" ?>
+{{!
+    For sitemap.xml, see https://www.sitemaps.org/protocol.html
+    For date/time format, see https://www.w3.org/TR/NOTE-datetime
+}}
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+    <url>
+        <loc>{{site.info.base_url}}/</loc>
+        <changefreq>daily</changefreq>
+        <priority>0.8</priority>
+    </url>
+    {{#each (sitemap site.pages) }}
```

Comment:
Optional: You might just leave a note after this block that says something like:

> When Bartholomew sees `(sitemap ...)`, it will look for a Rhai file named `sitemap.rhai` in the scripts directory, and will execute that script (above).

Reply:
Thanks Matt. Great suggestion. Added in most recent commit.